tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -537,18 +945,104 @@ public void testPartialIteratingTryWithResource() throws Exception {
         getRootPath(); // will fail on windows if iterator not closed correctly
     }
 
+    @Test
+    public void testMismatchSecretKey() throws Exception {
+        char[] actualPassword = "secretPassword".toCharArray();
+        PasswordFactory factory = PasswordFactory.getInstance(ClearPassword.ALGORITHM_CLEAR, ELYTRON_PASSWORD_PROVIDERS);
+        ClearPassword clearPassword = (ClearPassword) factory.generatePassword(new ClearPasswordSpec(actualPassword));
+        FileSystemSecurityRealm securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath())
+                .setLevels(2)
+                .setSecretKey(key)
+                .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                .build();
+        ModifiableRealmIdentity newIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
+        newIdentity.create();
+        newIdentity.setCredentials(Collections.singleton(new PasswordCredential(clearPassword)));
+        securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath(false))
+                .setLevels(2)
+                .setSecretKey(SecretKeyUtil.generateSecretKey(192))
+                .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                .build();
+        ModifiableRealmIdentity existingIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
+        try {
+            existingIdentity.verifyEvidence(new PasswordGuessEvidence(actualPassword));
+        } catch (Exception e) {
+            assertEquals("ELY13005: Filesystem-backed realm unable to decrypt identity", e.getCause().getMessage());
```

Comment:
Instead of referencing the id here, you could just do something like this:

`assertTrue(e.getCause().getMessage().contains("unable to decrypt identity")